ecosystem/typescript/sdk/src/utils/property_map_serializer.ts

```diff
@@ -0,0 +1,40 @@
+import assert from "assert";
+import {
+  AnyNumber,
+  bcsSerializeBool,
+  bcsSerializeStr,
+  bcsSerializeU128,
+  bcsSerializeU8,
+  bcsSerializeUint64,
+  Bytes,
+} from "../bcs";
+import { TypeTagParser } from "../transaction_builder";
+import { TypeTagAddress, TypeTagBool, TypeTagStruct, TypeTagU128, TypeTagU64, TypeTagU8 } from "../aptos_types";
+
+export function getPropertyValueRaw(values: Array<string>, types: Array<string>): Array<Bytes> {
+  assert(values.length === types.length);
+  const results = new Array<Bytes>();
+  types.forEach((typ, index) => {
+    const typeTag = new TypeTagParser(typ).parseTypeTag();
+    if (typeTag instanceof TypeTagBool) {
+      const res: boolean = JSON.parse(values[index]);
+      results.push(bcsSerializeBool(res));
+    } else if (typeTag instanceof TypeTagStruct && (typeTag as TypeTagStruct).isStringTypeTag()) {
+      results.push(bcsSerializeStr(values[index]));
+    } else if (typeTag instanceof TypeTagU8) {
+      const res: number = JSON.parse(values[index]);
+      results.push(bcsSerializeU8(res));
+    } else if (typeTag instanceof TypeTagU64) {
+      const res: AnyNumber = BigInt(values[index]);
+      results.push(bcsSerializeUint64(res));
+    } else if (typeTag instanceof TypeTagU128) {
+      const res: AnyNumber = BigInt(values[index]);
+      results.push(bcsSerializeU128(res));
+    } else if (typeTag instanceof TypeTagAddress) {
+      results.push(bcsSerializeStr(values[index]));
+    } else {
+      results.push(bcsSerializeStr(values[index]));
+    }
+  });
+  return results;
+}
```

Comment:
Here is what I meant
```
import {
  Bytes,
  Serializer,
} from "../bcs";
import { TypeTagParser } from "../transaction_builder";
import { serializeArg } from "../transaction_builder/builder_utils";

export function getPropertyValueRaw(values: Array<string>, types: Array<string>): Array<Bytes> {
  if (values.length !== types.length) {
    throw new Error("Invalid input.");
  }

  const results = new Array<Bytes>();
  types.forEach((typ, index) => {
    const typeTag = new TypeTagParser(typ).parseTypeTag();

    const serializer = new Serializer();
    serializeArg(values[index], typeTag, serializer);
    results.push(serializer.getBytes());
  });
  return results;
}

```
Also, we should not use `import assert from "assert"` as it is not working in Browser.  Although we can use it in tests (only node environment), we should use Jest's assertion libraries.

Reply:
done